xilem_web/src/after_update.rs

```diff
@@ -0,0 +1,207 @@
+// Copyright 2023 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use xilem_core::{MessageResult, Mut, View, ViewId, ViewMarker};
+
+use crate::{DomNode, DomView, DynMessage, ViewCtx};
+
+pub struct AfterBuild<E, F> {
+    element: E,
+    callback: F,
+}
+
+pub struct AfterRebuild<E, F> {
+    element: E,
+    callback: F,
+}
+
+pub struct BeforeTeardown<E, F> {
+    element: E,
+    callback: F,
+}
+
+/// This adds a callback to a DOM node
+/// that is invoked after it has been created.
+/// The callback has a reference to the raw DOM node as its only parameter.
+///
+/// Caution: At this point, however,
+/// no properties have been applied to the node.
+///
+/// The use of this function should be avoided and
+/// should only be utilized in exceptional cases!
+pub fn after_build<E, F>(element: E, callback: F) -> AfterBuild<E, F> {
+    AfterBuild { element, callback }
+}
+
+/// This adds a callback to a DOM node
+/// that is invoked after it has been rebuild.
+/// The callback has a reference to the raw DOM node as its only parameter.
+///
+/// The use of this function should be avoided and
+/// should only be utilized in exceptional cases!
+pub fn after_rebuild<E, F>(element: E, callback: F) -> AfterRebuild<E, F> {
+    AfterRebuild { element, callback }
+}
+
+/// This adds a callback to a DOM node
+/// that is invoked before it is destroyed.
+/// This adds a callback to the view
+/// that is called before a DOM node is destroyed.
+/// The callback has a reference to the raw DOM node as its only parameter.
+///
+/// The use of this function should be avoided and
+/// should only be utilized in exceptional cases!
```

Comment:
Maybe (and if you agree, probably in `after_build` and `after_rebuild` as well)?:

```suggestion
/// As accessing the underlying raw DOM node can mess with the inner logic of xilem_web,
/// this should only be used as an escape-hatch for properties not supported by xilem_web. E.g. to be interoperable with external javascript libraries.
```

btw. I think adding bounds here actually helps with documenting (and prevents misuse) what should be used as `element` and can be given to the `callback`

Reply:
You mean something like this?:

```rust
pub fn before_teardown<E, F, State>(element: E, callback: F) -> BeforeTeardown<E, F>
  where
    E: DomView<State>,
    F: Fn(&web_sys::Element) + 'static,
{
    BeforeTeardown { element, callback }
}
```